src/network/tc/qdisc.c

```diff
@@ -250,15 +252,23 @@ int config_parse_tc_qdiscs_parent(
 
         if (streq(rvalue, "root"))
                 qdisc->parent = TC_H_ROOT;
-        else if (streq(rvalue, "clsact"))
+        else if (streq(rvalue, "clsact")) {
                 qdisc->parent = TC_H_CLSACT;
-        else {
+                qdisc->handle = TC_H_MAKE(TC_H_CLSACT, 0);
+        } else if (streq(rvalue, "ingress")) {
+                qdisc->parent = TC_H_INGRESS;
+                qdisc->handle = TC_H_MAKE(TC_H_INGRESS, 0);
+        } else {
                 log_syntax(unit, LOG_ERR, filename, line, r,
                            "Failed to parse [QueueDiscs] 'Parent=', ignoring assignment: %s",
                            rvalue);
                 return 0;
         }
 
+        qdisc->tca_kind = strdup(rvalue);
```

Comment:
`free_and_strdup()`

Reply:
This is only for `clsact` and  `ingress` not for `root`. I fixed it  . `netem` , `tbq` etc as it is.